test/nftTrades.test.ts

```diff
@@ -184,103 +183,99 @@ describe('SwapRouter', () => {
-  //         permit2TransferFrom: true,
-  //         permit2Permit: {
-  //           ...permit2Data,
-  //           signature,
-  //         },
-  //       },
-  //     ]
-
-  //     const seaportTrade = new SeaportTrade([seaportData])
-  //     const methodParameters = SwapRouter.swapNFTCallParameters([seaportTrade])
-  //     const methodParametersV2 = SwapRouter.swapCallParameters(seaportTrade)
-  //     registerFixture('_SEAPORT_BUY_ITEMS_ERC20_PERMIT_NO_APPROVE', methodParameters)
-  //     expect(hexToDecimalString(methodParameters.value)).to.eq('0')
-  //     expect(methodParameters.calldata).to.eq(methodParametersV2.calldata)
-  //     expect(methodParameters.value).to.eq(hexToDecimalString(methodParametersV2.value))
-  //   })
-  // })
+  describe('SeaportV1_5', () => {
+    it('encodes buying two NFTs from Seaport v1.5 with ETH', async () => {
+      const seaportTrade = new SeaportTrade([seaportV1_5DataETH])
+      const value = seaportTrade.getTotalPrice(ETH_ADDRESS)
+      const methodParameters = SwapRouter.swapNFTCallParameters([seaportTrade])
+      const methodParametersV2 = SwapRouter.swapCallParameters(seaportTrade)
+      registerFixture('_SEAPORT_V1_5_BUY_ITEMS_ETH', methodParametersV2)
+      expect(hexToDecimalString(methodParameters.value)).to.eq(value.toString())
+      expect(methodParameters.calldata).to.eq(methodParametersV2.calldata)
+      expect(hexToDecimalString(methodParameters.value)).to.eq(methodParametersV2.value)
+    })
+  })
 
   describe('SeaportV1_4', () => {
+    const wallet = new Wallet(utils.zeroPad('0x1234', 32))
+
     it('encodes buying two NFTs from Seaport v1.4 with ETH', async () => {
       const seaportV1_4Trade = new SeaportTrade([seaportV1_4DataETH])
       const value = seaportV1_4Trade.getTotalPrice(ETH_ADDRESS)
       const methodParameters = SwapRouter.swapNFTCallParameters([seaportV1_4Trade])
       const methodParametersV2 = SwapRouter.swapCallParameters(seaportV1_4Trade)
-      registerFixture('_SEAPORT_V1_4_BUY_ITEMS_ETH', methodParameters)
+      registerFixture('_SEAPORT_V1_4_BUY_ITEMS_ETH', methodParametersV2)
       expect(hexToDecimalString(methodParameters.value)).to.eq(value.toString())
       expect(methodParameters.calldata).to.eq(methodParametersV2.calldata)
-      expect(methodParameters.value).to.eq(hexToDecimalString(methodParametersV2.value))
+      expect(hexToDecimalString(methodParameters.value)).to.eq(methodParametersV2.value)
+    })
+
+    it('encodes buying 1 NFT from Seaport with ERC20, with Permit and Approve', async () => {
+      // get the basic seaport data for ERC20 trade
+      let seaportData = seaportV1_4DataERC20
+
+      // make permit
+      const GALA_ADDRESS = '0x15D4c048F83bd7e37d49eA4C83a07267Ec4203dA'
+      const permit2Data = makePermit(GALA_ADDRESS, undefined, undefined, FORGE_ROUTER_ADDRESS)
+      const signature = await generatePermitSignature(permit2Data, wallet, 1, FORGE_PERMIT2_ADDRESS)
+      seaportData.inputTokenProcessing = [
```

Comment:
What is this new attribute `inputTokenProcessing`? Is it custom defined?

Reply:
My opinion is that generally, an SDK entity class like `SeaportTrade` should take in purely what the API returns, and have all processing / transformation of the data contained within itself. Like create the `inputTokenProcessing` in the constructor of SeaportTrade. Though curious what your thoughts on this are!

Reply:
We added it in PR #112 ... I kept the field separate so it could be like `items=API data` and then `inputTokenProcessing=other stuff` and its optional. That being said I'm happy to separate it out... dont have a strong opinion either way :)
Would probably do that in another PR though as this is something thats already in the SDK

Reply:
Hmm, I see yeah honestly I think its fine since we have to pass in the permitData anyways and that seems independent of the SDK entity.